.github/workflows/build-spack.yml

```diff
@@ -137,10 +137,22 @@ jobs:
           cmake --preset=${CMAKE_PRESET} --log-level=VERBOSE
       - name: Build all
         id: build
-        working-directory: build
+        working-directory: ${{format('{0}', matrix.special == 'clang-tidy' && github.workspace || 'build')}}
+        env:
+          BASE_REF: "${{format('{0}', github.base_ref || 'develop')}}"
         run: |
           if [ "${{matrix.special}}" == "clang-tidy" ]; then
-            ninja -k0
+            if [ "${{github.event_name}}" == "schedule" ]; then
+              ninja -k0
+              exit $?
+            fi
+            FILES=$(git diff --name-only --diff-filter=ACM "$BASE_REF" HEAD)
+            if [ -z "$FILES" ]; then
+              echo "No files to run clang-tidy on."
+              exit 0
+            fi
+            grep -E '^(src|app|example)/.*\.cc$' - <<< "$FILES" | xargs ${{env.CLANG_TIDY}} -p build
+            exit $?
```

Comment:
Instead of the special casing in here, how about we do a different `Run incremental clang-tidy` step and add  `if` blocks here and for that?  (We're not "building all"!)